# Chi-Square test 

- χ2=∑(O−E)2/E
- Expected Frequency = (Row_total)*(col_total)/Grand Total
### Degrees of freedom:
- df = (row-1)(column-1)

Objective:
To use the Chi-Square test for independence to determine if there's a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.

### Data Provided:
The data is summarized in a contingency table showing the counts of customers in each satisfaction level for both types of devices:
- Satisfaction	Smart Thermostat	Smart Light	Total
- Very Satisfied	50	70	120
- Satisfied	80	100	180
- Neutral	60	90	150
- Unsatisfied	30	50	80
- Very Unsatisfied	20	50	70
- Total	240	360	600


In [17]:
import numpy as np
import pandas as pd
from scipy.stats import chi2

Assignment Tasks:
1. State the Hypotheses:

H₀: Device type and customer satisfaction are independent.

H₁: Device type and customer satisfaction are associated.

2. Compute the Chi-Square Statistic:
- χ2=∑(O−E)2/E

In [2]:
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

In [ ]:
# Step 1: Compute chi-square manually
row_totals = observed.sum(axis=1).reshape(-1,1)
col_totals = observed.sum(axis=0).reshape(1,-1)
grand_total = observed.sum()

(row_totals,col_totals,grand_total)

(array([[120],
        [180],
        [150],
        [ 80],
        [ 70]]),
 array([[240, 360]]),
 600)

In [ ]:
# Expected frequencies
expected = row_totals @ col_totals / grand_total
expected


array([[ 48.,  72.],
       [ 72., 108.],
       [ 60.,  90.],
       [ 32.,  48.],
       [ 28.,  42.]])

In [13]:
chi_square_stat = ((observed - expected)**2 / expected).sum()
chi_square_stat

5.638227513227513

#### 3. Determine the Critical Value:
Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)
- df = (row-1)(column-1)

In [15]:
observed.shape

(5, 2)

In [14]:
dof = (observed.shape[0]-1) * (observed.shape[1]-1)
dof

4

In [16]:
# ---- Step 5: Critical value & p-value ----
alpha = 0.05
chi_critical = chi2.ppf(1-alpha, dof)
p_value = 1 - chi2.cdf(chi_square_stat, dof)

(chi_critical,p_value)

(9.487729036781154, 0.22784371130697179)

### 4. Make a Decision:
Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.

# Since, 𝜒2=5.64<9.488, we fail to reject H₀.

# Final Conclusion

- There is no significant association between device type and customer satisfaction.
- Satisfaction does not depend on whether a customer bought a thermostat or a light.